app/IOSGateway.h

```diff
@@ -0,0 +1,24 @@
+//
+//  IOSGateway.h
+//  iSH
+//
+//  Created by Miguel Vanhove on 15/02/2019.
+//
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface IOSGateway : NSObject
+
++ (IOSGateway *_Nullable ) sharedSession;
+
+- (void)setup;
+- (BOOL)handleOpenURL:(NSURL *)url;
```

Comment:
I think this method would be better served being named `-(BOOL)canHandleOpeningURL:(NSURL *)url`.

Reply:
why not ? More clear with the canHandleOpeningURL :)